core/src/main/java/org/apache/calcite/rel/metadata/MetadataDef.java

```diff
@@ -41,7 +42,10 @@ private MetadataDef(Class<M> metadataClass,
     this.handlerClass = handlerClass;
     this.methods = ImmutableList.copyOf(methods);
     final Method[] handlerMethods = Arrays.stream(handlerClass.getDeclaredMethods())
-        .filter(m -> !m.getName().equals("getDef")).toArray(i -> new Method[i]);
+        .filter(m -> !m.getName().equals("getDef"))
```

Comment:
@jacques-n it seems this filter was originally introduced via CALCITE-4929 (see [4ff5fa8](https://github.com/apache/calcite/commit/4ff5fa89c180ebc30d8fb324c2a50b0e9797b9ca) ), what do you think about the current patch?
Also, I noticed that in  CALCITE-4929 a similar filter was introduced in [RelMetadataHandlerGeneratorUtil.java](https://github.com/apache/calcite/commit/4ff5fa89c180ebc30d8fb324c2a50b0e9797b9ca#diff-b5bac3d2a905799f710aef8e5fd02dd659dc9e6e2c8e7b05ed0c34022025b8e1) , if we apply this patch, should we apply the change in both files too?

Reply:
Yes, we should apply the same logic in both places. I was thinking that we should probably extract this to a utility method as opposed to continue duplication. As part of this, let's also add specific units tests which verify the utility method. You up for that @rpuch?

Reply:
@jacques-n sure, I'll do it. Thank you guys for your suggestions.

Reply:
Done